<?xml version="1.0"?>

<launch>

  <arg name="world_name" default=""/>
  
    <!--模型车的位置不能修改-->
  <arg name="x_pos" default="-0.5"/>
  <arg name="y_pos" default="0"/>
  <arg name="z_pos" default="0.0"/>

  <env name="GAZEBO_MODEL_PATH" value="$GAZEBO_MODEL_PATH:$(find racecar_gazebo)/models"/>
  <env name="GAZEBO_RESOURCE_PATH" value="$GAZEBO_RESOURCE_PATH:$(find racecar_gazebo)/models"/>

  <!-- 启动gazebo，加载.world文件 -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find racecar_gazebo)/worlds/$(arg world_name).world"/>
    <arg name="paused" default="false"/>
    <arg name="use_sim_time" default="true"/>
    <arg name="gui" default="true"/>
    <arg name="headless" default="false"/>
    <arg name="debug" default="false"/>
  </include>

  <!-- 加载机器人.urdf文件，启动robot_state_publisher节点发布机器人的tf信息 -->
  <include file="$(find racecar_description)/launch/racecar_spawn.launch">
    <arg name="x_pos" default="$(arg x_pos)"/>
    <arg name="y_pos" default="$(arg y_pos)"/>
    <arg name="z_pos" default="$(arg z_pos)"/>
  </include>

  <!-- ros_control racecar launch file -->
  <!-- 加载gazebo中的机器人joint控制节点 -->
  <include file="$(find racecar_gazebo)/launch/gazebo_control.launch"/>

</launch>
